python/vtool/maya_lib/api.py

```diff
@@ -1452,427 +1452,426 @@ def multiply_matrix(matrix1, matrix2):
-                    
+
                     index =  surf_it.uvIndices()
-                    
+
                     indices.append([index[1], index[0]])
-                    
+
                     surf_it.next()
                     if inc2 == 10:
                         break
-                
+
                 surf_it.nextRow()
-                
+
                 if inc == 10:
                     break
-            
+
             """
             double_component = om.MFnDoubleIndexedComponent()
             components = double_component.create(om.MFn.kSurfaceCVComponent)
             double_component.addElements(indices)
-            
+
         if api_type == om.MFn.kLattice:
             dagpath = om.MDagPath.getAPathTo(output_object)
-            
+
             lattice_name = dagpath.fullPathName()
-            
+
             pts = cmds.ls('%s.pt[*][*][*]' % lattice_name, flatten = True)
-            
+
             indices = []
-            
+
             for pt in pts:
                 numbers = util.get_square_bracket_numbers(pt)
                 indices.append(numbers)
-            
+
             triple_component = om.MFnTripleIndexedComponent()
             components = triple_component.create(om.MFn.kLatticeComponent)
             triple_component.addElements(indices)
-        
+
         return dagpath, components
-        
-        
-        
+
+
+
 
 def get_components(indices):
 
     fn_comp_verts = om.MFnSingleIndexedComponent()
     components = fn_comp_verts.create(om.MFn.kMeshVertComponent)
     fn_comp_verts.addElements(indices)
-    
+
     return components
-    
+
 
 def set_skin_weights(skin_cluster, weights = 0, index = 0, components = None, influence_array = None):
-    
+
     skin_object = get_object(skin_cluster)
     dag_path, component = get_skin_components(skin_cluster, index)
     skin_fn = omAnim.MFnSkinCluster(skin_object)
-    
-    if type(components) == list or type(components) == tuple:
+
+    if isinstance(components, list) or isinstance(components, tuple):
         components = get_components(components)
     if not components:
         components = component
-    
+
     influence_count = 0
-    
+
     if influence_array:
         influence_count = len(influence_array)
-        
+
         int_influence_array = om.MIntArray()
-        
+
         for inc in range(0,influence_count):
             int_influence_array.append(influence_array[inc])
-        
+
         influence_array = int_influence_array
-    
+
     if not influence_array:
         influence_dag_paths = skin_fn.influenceObjects()
         influence_count = len(influence_dag_paths)
         influence_array =  om.MIntArray()
-    
+
         for inc in range(0,influence_count):
-            
+
             index = skin_fn.indexForInfluenceObject(influence_dag_paths[inc])
             influence_array.append(index)
-          
-    
+
+
     weight_array = None
-    if type(weights) == type(om.MDoubleArray()):

```

Comment:
V: Error!	Traceback (most recent call last):	
V:		  File "E:/dev/git/vtool/python\vtool\util.py", line 76, in wrapper	
V:		    return_value = function(*args, **kwargs)	
V:		  File "E:/dev/git/vtool/python\vtool\data.py", line 1673, in import_data	
V:		    self._import_maya_data(filepath, selection)	
V:		  File "E:/dev/git/vtool/python\vtool\data.py", line 1323, in _import_maya_data	
V:		    result = self.import_skin_weights(folder_path, mesh, first = first)	
V:		  File "E:/dev/git/vtool/python\vtool\data.py", line 1568, in import_skin_weights	
V:		    maya_lib.api.set_skin_weights(skin_cluster, weight_array, 0)	
V:		  File "E:/dev/git/vtool/python\vtool\maya_lib\api.py", line 1837, in set_skin_weights	
V:		    if isinstance(weights, om.MDoubleArray()):	
V:		TypeError: isinstance() arg 2 must be a type or tuple of types

Reply:
The error above I get from Maya when running the human body template
Maybe isinstance doesn't work on om.MDoubleArray()